src/equations/bbm_1d.jl

```diff
@@ -0,0 +1,196 @@
+# TODO: Use this nondimensionalized version or use dimensional version including gravity and depth?
+@doc raw"""
+    BBMEquation1D(; bathymetry_type = bathymetry_flat, eta0 = 0.0)
+
+BBM (Benjamin–Bona–Mahony) equation in one spatial dimension. The equations are given by
+```math
+\begin{aligned}
+  \eta_t + \eta\eta_x + \eta_x - \eta_{xxt} &= 0.
```

Comment:
I use $\eta$ here instead of $u$ as in the main reference for the BBM equation for consistency with the other equations.

Reply:
👍